SmartDeviceLink/private/SDLMenuManager.m

```diff
@@ -560,8 +564,11 @@ - (BOOL)sdl_isDynamicMenuUpdateActive:(SDLDynamicMenuUpdatesMode)dynamicMenuUpda
 - (BOOL)sdl_shouldRPCsIncludeImages:(NSArray<SDLMenuCell *> *)cells {
     for (SDLMenuCell *cell in cells) {
         SDLArtwork *artwork = cell.icon;
+        SDLArtwork *secondaryArtwork = cell.secondaryArtwork;
         if (artwork != nil && !artwork.isStaticIcon && ![self.fileManager hasUploadedFile:artwork]) {
             return NO;
+        } else if (secondaryArtwork != nil && !secondaryArtwork.isStaticIcon && ![self.fileManager hasUploadedFile:secondaryArtwork]) {
```

Comment:
This check fails if:
1. `[self.windowCapability hasImageFieldOfName:SDLImageFieldNameMenuCommandSecondaryImage]` or `[self.windowCapability hasImageFieldOfName:SDLImageFieldNameMenuSubMenuSecondaryImage]` returns `false`
2. One of the menu `artwork`s is a static icon OR one of the menu `artwork`s is `nil` so the first `if` is skipped.
3. The same menu item's `secondaryArtwork` is not a static icon but since it was not uploaded due to the system not supporting `secondaryArtwork`s, this `else if` is selected.
 
This means the menu is uploaded with no images even though some of the menu items have `artwork`. You should add a check if the `secondaryArtwork` is supported on the module before checking if the file manager has successfully uploaded the `secondaryArtwork`